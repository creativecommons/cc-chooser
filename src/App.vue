<template>
    <div id="app">
        <Header :title="$t('chooser')"/>
        <Feedback/>
        <div class="container" id="site-container">
            <Chooser v-model="selected"/>
            <hr>
            <HelpSection/>
            <hr>
            <HTMLGenerator :shortLicenseName="selected.shortName"/>
        </div>
        <Footer>
            <div class="panel">
               <div>
                   <p>
                       Built with <a href="https://creativecommons.github.io/cc-vocabulary/">Vocabulary</a>.
                   </p>
                   <p>
                       <a href="https://github.com/creativecommons/cc-chooser">Icons</a> by The Noun Project and FontAwesome.
                   </p>
                   <p>
                       View this page's source code <a href="https://github.com/creativecommons/cc-chooser">on Github</a>.
                   </p>
                </div>
                <Locale/>
            </div>
        </Footer>
    </div>
</template>
<script>
// TODO Reduce custom styling in favour of Vocabulary styles
import '@creativecommons/vocabulary/root.css'
import '@creativecommons/vocabulary/vocabulary.css'

import Chooser from './components/Chooser'
import HelpSection from './components/HelpSection'
import HTMLGenerator from './components/HTMLGenerator'
import { Header, Footer, LicenseIconography, Locale } from '@creativecommons/vocabulary'
import Feedback from './components/Feedback'

export default {
    name: 'App',
    components: {
        Chooser,
        HelpSection,
        HTMLGenerator,
        Header,
        Footer,
        LicenseIconography,
        Locale,
        Feedback
    },
    data() {
        return {
            selected: {
                shortName: 'CC BY 4.0',
                fullName: 'Attribution 4.0 International'
            }
        }
    },
    methods: {
        onLicenseUpdate(e) {
            this.selected = e
        }
    }
}
</script>

<style>
@import url("https://fonts.googleapis.com/css?family=Source+Sans+Pro%3A400%2C600%2C700%7CRoboto+Condensed&ver=4.9.8");

#app {
    -webkit-font-smoothing: antialiased;
    -moz-osx-font-smoothing: grayscale;
    color: #2c3e50;
}

hr {
    margin-left: 5%;
    margin-right: 5%;
}

#site-container {
    padding-top: 2rem;
    padding-bottom: 2%;
}

.desktop-hide { display: none; }
.desktop-show { display: block; }

@media only screen and (max-width: 1025px) {
    #site-container {
        margin-left: 2%;
        margin-right: 2%;
    }

    .mobile-hide { display: none; }
    .mobile-show { display: block; }
}

@media only screen and (max-width: 670px) {
    #site-container {
        margin-left: 3%;
        margin-right: 3%;
    }
}
.vocab-container .vocab-grid .panel {
 height: 100%;
    display: -webkit-box;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-orient: vertical;
    -webkit-box-direction: normal;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-box-pack: justify;
    -ms-flex-pack: justify;
    justify-content: space-between;
    }
    .vocab.select-field.normal-sized {
        font-size: 1rem;
    }
    .vocab.select-field>.field.has-addons {
        padding-left: calc(1em + var(--select-field-addons-space));
        margin-bottom: 0;
        min-width: 10em;
    }
    .vocab.select-field>.field>option[disabled] {
        opacity: 0.6;
    }

.panel {
    margin-top: 1rem;
}
</style>
<style  scoped>
    p { margin-bottom: .5em; }
    #custom .panel > div:first-child {
        margin-bottom: 10vh;
    }
    .vocab.footer {
        font-family: Source Sans Pro,Noto Sans,Arial,Helvetica Neue,Helvetica,sans-serif;
        line-height: 1.2;
    }
</style>

<i18n src="./locales/App.json">
</i18n>
