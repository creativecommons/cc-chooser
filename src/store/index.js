import Vue from 'vue'
import Vuex from 'vuex'
import { defaultAttributes, CC0Attributes, attrToShort, attrToFull, licenseUrl, licenseIconsArr } from '../utils/license-utilities.js'

Vue.use(Vuex)

const defaultState = {
    currentLicenseAttributes: { ...defaultAttributes },
    attributionDetails: {
        creatorName: '',
        creatorProfileUrl: '',
        workTitle: '',
        workUrl: ''
    }
}

const createStore = (state) => {
    const initialState = { ...defaultState, ...state }
    return new Vuex.Store({
        state: initialState,
        getters: {
            isLicenseSelected: state => {
                /**
                 * By default, all four license attributes are undefined
                 * As soon as the first attribute(BY) is selected (true/false),
                 * we can show the recommended license
                 */
                return state.currentLicenseAttributes.BY !== undefined
            },
            shortName: state => {
                return attrToShort(state.currentLicenseAttributes)
            },
            fullName: state => {
                return attrToFull(state.currentLicenseAttributes)
            },
            licenseUrl: state => (mode) => {
                return licenseUrl(state.currentLicenseAttributes, mode)
            },
            iconsList: state => {
                return licenseIconsArr(state.currentLicenseAttributes)
            }
        },
        mutations: {
            /**
             * Updates current license attributes when user selects radio option.
             * Edge case: If user selects ND, SA should be set to false
             * @param state
             * @param {string} stepName
             * @param {Boolean} isSelected
             */
            setSelected(state, { stepName, isSelected }) {
                if (stepName === 'ND' && isSelected && state.currentLicenseAttributes.SA) {
                    state.currentLicenseAttributes = {
                        ...state.currentLicenseAttributes,
                        SA: false,
                        ND: true
                    }
                } else if (['BY', 'NC', 'ND', 'SA'].indexOf(stepName) > -1) {
                    state.currentLicenseAttributes = {
                        ...state.currentLicenseAttributes,
                        [stepName]: isSelected
                    }
                }
            },
            updateAttributesFromShort(state, shortName) {
                if (shortName.includes('CC0')) {
                    state.currentLicenseAttributes = { ...CC0Attributes }
                } else {
                    state.currentLicenseAttributes.BY = true
                    state.currentLicenseAttributes.NC = !!shortName.includes('NC')
                    state.currentLicenseAttributes.ND = !!shortName.includes('ND')
                    state.currentLicenseAttributes.SA = !!shortName.includes('SA')
                }
            },
            setCreatorName(state, newName) {
                state.attributionDetails.creatorName = newName
            },
            setCreatorProfileUrl(state, newName) {
                state.attributionDetails.creatorProfileUrl = newName
            },
            setWorkTitle(state, newName) {
                state.attributionDetails.workTitle = newName
            },
            setWorkUrl(state, newName) {
                state.attributionDetails.workUrl = newName
            }
        }
    })
}

export default createStore
