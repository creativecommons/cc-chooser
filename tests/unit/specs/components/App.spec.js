import { shallowMount, createLocalVue } from '@vue/test-utils'
import App from '@/App'
import Buefy from 'buefy'
import VueI18n from 'vue-i18n'
import Vuex from 'vuex'
import Vue from 'vue'
import store from '@/store'

describe('App.vue', () => {
    let wrapper
    let localVue

    // Always creates a shallow instance of component
    beforeEach(() => {
        localVue = createLocalVue()
        localVue.use(Vuex)
        localVue.use(Buefy)
        Vue.use(VueI18n)
        wrapper = shallowMount(App, {
            store,
            localVue,
            data() {
                return {
                    currentStepId: 7,
                    showLicense: false,
                    shouldShake: false,
                    windowWidth: window.innerWidth
                }
            },
            mocks: {
                $t: key => key
            }
        })
        wrapper.vm.$on('input', (newVal) => {
            wrapper.setProps({ value: newVal })
        })
    })

    it('Check that all computed i18n props return correct values after true selected', () => {
        expect(wrapper.vm.showLicenseUse).toBe(true)
    })
})
