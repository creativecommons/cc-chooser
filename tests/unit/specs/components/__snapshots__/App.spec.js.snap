// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.vue Check if the App.vue component has the expected UI 1`] = `
<div id="app">
  <header-section-stub></header-section-stub>
  <div id="site-container" class="container">
    <nav aria-label="breadcrumbs" class="breadcrumb caption bold">
      <ul>
        <li><a href="#">Home</a></li>
        <li><a href="#">Child Page</a></li>
        <li class="is-active"><a href="#" aria-current="page">Chooser</a></li>
      </ul>
    </nav>
    <div class="page-head">
      <div class="select-license-column">
        <h2 class="vocab">
          LICENSE CHOOSER
        </h2>
        <p class="stepper-instructions vocab-body body-bigger">
          Follow the steps to select the appropriate license for your work. This site does not store any information.
        </p>
      </div>
      <localechooser-stub class="locale-chooser"></localechooser-stub>
    </div>
    <div class="columns">
      <div class="column">
        <stepper-stub value="0"></stepper-stub>
        <help-section-stub></help-section-stub>
      </div>
      <div class="column">
        <div class="fixed-right-column">
          <!---->
          <!---->
        </div>
      </div>
    </div>
  </div>
  <footer-section-stub></footer-section-stub>
</div>
`;
